addons/website/static/src/snippets/s_image_gallery/options.js

```diff
@@ -286,6 +286,15 @@ options.registry.gallery = options.Class.extend({
         this.$target.off('slide.bs.carousel').off('slid.bs.carousel');
         this.$('li.fa').off('click');
     },
+    /**
+     * Gets classNames from data and selects the right class depending on range value.
+     *
+     * @see this.selectClass for parameters
+     */
+    setSpacing(previewMode, widgetValue, params) {
+        params.possibleValues = JSON.parse(params.classNames);
+        this.selectClass(previewMode, params.possibleValues[+widgetValue], params);
+    },
```

Comment:
Let's make this whole commit better. This was fine but feels kinda hackish for an option method to call other option methods with tweaked params.

What would be nice is for the range widget to be reusable to do this kind of stuff. Adapt the we-range widget to accept a "values" data so here it can be defined:
```xml
<we-range string="Images Spacing" data-select-class="" data-values="[a, b, c, d]"/>
```
(with a, b, c, d being the classes).
Then the we-range should automatically know it is a slider from 0 to 3 and it should not notify number values anymore but the associated value of data-values.

That way, you don't have to do any change in the option here: data-select-class is the method you use, it works out of the box with the values it is given to. Only the range widget has to be adapted to make this feature work. The feature is the same here: just the widget to use is different = only the widget should be changed.